packages/react-sdk/src/wrappers/LivestreamPlayer/LivestreamPlayer.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react';
+import { Call } from '@stream-io/video-client';
+import { useStreamVideoClient } from '@stream-io/video-react-bindings';
+import {
+  LivestreamLayout,
+  LivestreamLayoutProps,
+  StreamCall,
+} from '../../core';
+
+export type LivestreamPlayerProps = {
+  /**
+   * The call type. Usually `livestream`.
+   */
+  callType: string;
+  /**
+   * The call ID.
+   */
+  callId: string;
+  /**
+   * The props for the {@link LivestreamLayout} component.
+   */
+  layoutProps?: LivestreamLayoutProps;
+};
+
+export const LivestreamPlayer = (props: LivestreamPlayerProps) => {
+  const { callType, callId, layoutProps } = props;
+  const client = useStreamVideoClient();
+  const [call, setCall] = useState<Call>();
```

Comment:
BTW, I always wondered, why we don't recommend this in the tutorial:

```js
const call = useMemo(
  () => client.call(callType, callId),
  [client, callType, callId]
);
useEffect(() => {
  call.join().catch(/* ... */);
  return () => call.leave().catch(/* ... */);
}, [call]);
```

Feels more straightforward, plus saves one render.

Reply:
Oh, lol, didn't notice @santhoshvai's comment above saying the exact same thing!

Reply:
hey @myandrienko, @santhoshvai,

This combination fixes the problem with StrictMode but IMO it makes it a bit harder to work with the rest of the SDK as there isn't a way to "throw-away" the `call` instance anymore.

With `useState`, I can set the `call` to `undefined` once I don't need it anymore. This can't be done with `useMemo`. At least, not without re-inventing `useState` on top of `useMemo`.

`useMemo` will force users to do this:
```tsx
const MyApp = () => {
   const call = useMemo();
   useEffect(() => {}, []);
   return (
     <StreamCall call={call}>
        <MyExtraComponent />
     </StreamCall>
   );
}

const MyExtraComponent = () => {
  const { useCallCallingState } = useCallStateHooks();
  const callingState = useCallCallingState();
  
  const inCall = callingState !== CallingState.IDLE && callingState !== CallingState.LEFT;
  
  if (!inCall) return null;
  return <MyUI />;
}
```

instead of:
```tsx
const [call, setCall] = useState();
useEffect(() => {}, [...]);

if (!call) return null;
return <MyUi />;
```

Checking whether one has a valid `call` instance would be more complex as one would always have to peek into the `call.state` instead of doing simple `if (!call) {}` checks.

IMO, this is a good tradeoff for one extra render :)
What do you think about this?

Reply:
I feel like:
```tsx
const call = useMemo(() => client?.call(callType, callId), [client, callId, callType]);
```
should make the `call` variable either `Call` or `undefined` at which point you can definitelly provide a check for:
```tsx
!call
```
right? Change of state (either `client`, `callId` or `callType`) should esentially call a cleanup function with the instance of that scope and provide a new instance to begin "connecting" to. 

Unless client is always available and your current setup should be setting the `call` when the actuall call connects? Like this:

```tsx
 const [call, setCall] = useState<Call>();
  useEffect(() => {
    if (!client) return;
    const myCall = client.call(callType, callId);
    myCall.join().then(() => {
      setCall(myCall);
    }).catch((e) => {
      console.error('Failed to join call', e);
    });
    return () => {
      myCall.leave().catch((e) => {
        console.error('Failed to leave call', e);
      });
      setCall(undefined);
    };
  }, [callId, callType, client]);
```

Reply:
> now what? how can I set the `call` to `undefined` without an additional state property?|

I don't think you have to do anything as the new `call`  is coming in right away - correct?
The whole setup would look something like:

```ts
const call = useMemo(() => {
  if (!client || !type || !id) return undefined;
  return client.call(type, id);
}, [client, type, id]);

useEffect(() => {
  if (!call) return;

  const joinPromise = call.join().catch(console.error);

  return () => {
    joinPromise.then(() => call.leave());
  };
});
```

Reply:
Or maybe there's a some form of a limitation in terms of state that I'm forgetting but this should work just fine I think.

Reply:
got you here, useMemo means you can avoid a rerender, but you cant drop the call on leave easily